File: careerforge-frontend/careerforge-frontend/src/services/projectService.js
import axiosConfig from "../api/axiosConfig";

export default {

    getAll: () => axiosConfig.get("/projects"),

    create: (data) => axiosConfig.post("/projects", data),

    update: (id, data) => axiosConfig.put(`/projects/${id}`, data),

    delete: (id) => axiosConfig.delete(`/projects/${id}`)

};

File: careerforge-backend/src/main/java/com/careerforge/service/ProjectService.java
package com.careerforge.service;

import com.careerforge.entity.Project;
import java.util.List;

public interface ProjectService {
    Project create(Project project);
    List<Project> getAll();
    Project getById(Long id);
    Project update(Long id, Project project);
    void delete(Long id);
}